pkg/cmd/provisioning/nutanix/create_shared_secrets_test.go

```diff
@@ -0,0 +1,199 @@
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				credentialsDir, err := ioutil.TempDir(os.TempDir(), testCredentialsDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials")
+				credentialsSourceFilepath = testBasicAuthCredentials(t, "username", "password", credentialsDir)
+				return
+			},
+			verify: func(t *testing.T, manifestsDir string) {
+				files, err := ioutil.ReadDir(manifestsDir)
+				require.NoError(t, err, "unexpected error listing files in manifestsDir")
+				assert.Equal(t, 1, countNonDirectoryFiles(files), "Should be exactly one files in manifestsDir when one CredReq to process")
+			},
+			expectedErr: "",
+		},
+		{
+			name: "Non-existent source credentials file",
+			setup: func(t *testing.T) (credReqDir, targetDir, credentialsSourceFilepath string) {
+				credReqDir, err := ioutil.TempDir(os.TempDir(), testCredReqDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				targetDir, err = ioutil.TempDir(os.TempDir(), testTargetDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				credentialsSourceFilepath = "__does_not_exist__"
+				return
+			},
+			verify: func(t *testing.T, manifestsDir string) {
+				files, err := ioutil.ReadDir(manifestsDir)
+				require.NoError(t, err, "unexpected error listing files in manifestsDir")
+				assert.Zero(t, countNonDirectoryFiles(files), "Should be no files in manifestsDir when no CredReqs to process")
+			},
+			expectedErr: "The source credentials file __does_not_exist__ does not exist",
+		},
+		{
+			name: "Non-existent credentials requests directory",
+			setup: func(t *testing.T) (credReqDir, targetDir, credentialsSourceFilepath string) {
+				credReqDir = "__does_not_exist__"
+
+				targetDir, err := ioutil.TempDir(os.TempDir(), testTargetDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials requests")
+
+				credentialsDir, err := ioutil.TempDir(os.TempDir(), testCredentialsDirPrefix)
+				require.NoError(t, err, "Failed to create temp directory for credentials")
+				credentialsSourceFilepath = testBasicAuthCredentials(t, "username", "password", credentialsDir)
+				return
+			},
+			verify: func(t *testing.T, manifestsDir string) {
+				files, err := ioutil.ReadDir(manifestsDir)
+				require.NoError(t, err, "unexpected error listing files in manifestsDir")
+				assert.Zero(t, countNonDirectoryFiles(files), "Should be no files in manifestsDir when no CredReqs to process")
+			},
+			expectedErr: "Failed to process files containing CredentialsRequests: open __does_not_exist__: no such file or directory",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			credReqDir, targetDir, credentialsSource := tt.setup(t)
+			defer os.RemoveAll(credReqDir)
+			defer os.RemoveAll(targetDir)
```

Comment:
do we also need to add `defer os.RemoveAll(credentialsSource)`?

Reply:
Good catch! Added it.